cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,396 @@
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			expectError: false,
+		},
+		{
+			name: "Missing_ClusterDeployment",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{},
+			},
+			expectError: true,
+		},
+		{
+			name: "SyncSet_Failure",
+			cdList: &hivev1.ClusterDeploymentList{
+				Items: []hivev1.ClusterDeployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster",
+						Namespace: "uhc-production-test",
+					},
+				}},
+			},
+			csList: &v1alpha1.ClusterSyncList{
+				Items: []v1alpha1.ClusterSync{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cluster-sync",
+						Namespace: "uhc-production-test",
+					},
+					Status: v1alpha1.ClusterSyncStatus{
+						SyncSets: []v1alpha1.SyncStatus{{
+							Name:           "sync-failure",
+							Result:         "Failure",
+							FailureMessage: "Some error occurred",
+						}},
+					},
+				}},
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			if err := hivev1.AddToScheme(scheme); err != nil {
+				t.Fatalf("Failed to add ClusterDeployment to scheme: %v", err)
+			}
+
+			if err := v1alpha1.AddToScheme(scheme); err != nil {
+				t.Fatalf("Failed to add ClusterSync to scheme: %v", err)
+			}
+
+			// Add ClusterDeployments and ClusterSyncs to the fake client
+			objects := []client.Object{}
+			for _, cd := range tt.cdList.Items {
+				objects = append(objects, &cd)
+			}
+			for _, cs := range tt.csList.Items {
+				objects = append(objects, &cs)
+			}
+
+			client := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				Build()
+
+			o := &clusterSyncFailuresOptions{
+				kubeCli:   client,
+				clusterID: "test",
+			}
+
+			err := o.printFailingCluster()
+			if tt.expectError {
+				assert.Error(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+
+		})
+	}
+}
+
+func testListFailingClusterSyncs(t *testing.T, returnErr error, expectedResults []failingClusterSync, isEmpty bool) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockClient := mockk8s.NewMockClient(mockCtrl)
+
+	setupMockClient(mockClient, returnErr, isEmpty)
+
+	options := &clusterSyncFailuresOptions{
+		kubeCli: mockClient,
+	}
+
+	result, err := options.listFailingClusterSyncs()
+
+	if returnErr != nil {
+		assert.Error(t, err)
+		assert.Nil(t, result)
+	} else {
+		assert.NoError(t, err)
+		if isEmpty {
+			// If isEmpty is true, the result should be an empty slice
+			assert.Len(t, result, 0)
+		} else {
+			assert.NotNil(t, result)
+			assert.Len(t, result, len(expectedResults))
+			for i, expected := range expectedResults {
+				assert.Equal(t, expected.Name, result[i].Name)
```

Comment:
The `expectedResult` is only ever going to be length 1, so can we either add more results to the slice or remove this `for` loop?

Reply:
Removed the for loop.